python/ray/tune/trial_runner.py

```diff
@@ -392,12 +396,15 @@ def has_resources(self, resources):
         return self.trial_executor.has_resources(resources)
 
     def _stop_experiment_if_needed(self):
-        """Stops all trials if the user condition is satisfied."""
-
-        if self._stopper.stop_all() or self._should_stop_experiment:
+        """Stops all trials."""
+        fail_fast = self._fail_fast and self._has_errored
+        if (self._stopper.stop_all() or fail_fast
+                or self._should_stop_experiment):
             self._search_alg.set_finished()
-            [self.trial_executor.stop_trial(t) for t in self._trials]
-            logger.info("All trials stopped due to ``stopper.stop_all``.")
+            [
+                self.trial_executor.stop_trial(t) for t in self._trials
+                if t.status is not Trial.ERROR
```

Comment:
nit
```suggestion
                if t.is_finished()
```
might read more clearly

Reply:
I think it's actually not the same; you need to terminate paused ones too.